### PR TITLE
Tweak: SVG - Added more tags and attributes to the allowed list [ED-6777]

### PR DESCRIPTION
**Pull request by filipkis**
## PR Checklist
- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* adding more safe tags to SVG sanatization whitelist

## Description
An explanation of what is done in this PR

* Current whitelist of SVG tags and attributes was rather small and excluded many safe attributes (e.g. most SVG filters). This PR borrows the list from SVG sanitization library that includes a much wider set of tags and attributes.

## Test instructions
This PR can be tested by following these steps:

* create a svg that uses `feColorMatrix` or `feBlend` or any other such filter and try to upload it through elementor. Before this fix those filters would be removed and resulting SVG would be different. After this fix these are kept and the resulting SVG looks the same.

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

**Comment by rami-elementor** [comment]
What is your source of tags/attributes?

**Comment by filipkis** [comment]
I've taken the list from the following project: https://github.com/darylldoyle/svg-sanitizer

I'd be open to adjusting it to any other source, but the current one is really very limited and breaks all of our SVGs which we generate by exports from Figma.

File: core/files/file-types/svg.php [hunk 1]
```diff
@@ -450,94 +450,182 @@ private function has_js_value( $value ) {
...
-			'xmlns:xlink',
-			'xml:space',
-			'method',
-			'spacing',
-			'startoffset',
-			'dx',
-			'dy',
-			'rotate',
-			'textlength',
+			'accent-height',
```
Comment (matipojo) [inline core/files/file-types/svg.php:453]: Hi @filipkis 
Thanks for sending the list!
- Can you please add the credit for the list in the code?
- Can you fix the indentation of the list?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/files/file-types/svg.php [hunk 2]
```diff
@@ -569,40 +657,76 @@ private function get_allowed_attributes() {
...
+			'animatetransform',
 			'circle',
 			'clippath',
 			'defs',
-			'style',
 			'desc',
 			'ellipse',
-			'fegaussianblur',
 			'filter',
-			'foreignobject',
```
[the remark this answers is not in the document]
Comment (rami-elementor) [reply core/files/file-types/svg.php:582]: The `style` tag should not be removed. It's a valid SVG tag. I personally use the `style` tag in SVG files to add CSS animations.

[the remark this answers is not in the document]
Comment (rami-elementor) [reply core/files/file-types/svg.php:582]: @matipojo

* `style` moved to line 594
* `fegaussianblur` moved to line 719.
* `foreignobject` deleted(?)

File: core/files/file-types/svg.php (hunk 1 above)
Comment (matipojo) [reply core/files/file-types/svg.php:453]: @filipkis Any update?

Comment (filipkis) [reply core/files/file-types/svg.php:455]: Added the credits and fixed the indentation.

**Comment by rami-elementor** [comment]
@filipkis It's a good PR and we want to merge the code. But we see that you deleted some of the tags and attributes.

Can you please restore the deleted tags/attributes?

**Comment by rami-elementor** [comment]
Closing...

**Pull request closed without merging**